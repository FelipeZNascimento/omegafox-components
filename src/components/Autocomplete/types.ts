export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
