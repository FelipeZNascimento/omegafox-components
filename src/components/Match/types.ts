import { TClockAmericanFootball, TClockFootball, ITeamProps } from '../types';

export type TSport = 'football' | 'am_football';
export type TBetValues = 5 | 3 | 2 | 0;
export type TBET_VALUES = {
  FULL: TBetValues;
  HALF: TBetValues;
  MINIMUN: TBetValues;
  MISS: TBetValues;
};

export interface IMatchProps {
  bet?: number | null;
  betValue?: TBetValues | null;
  clock: TClockFootball | TClockAmericanFootball;
  expandableContent?: () => JSX.Element;
  id: number;
  isBigLogo?: boolean;
  isEditable?: boolean;
  isError?: boolean;
  isExpandable?: boolean;
  isForceMobile?: boolean;
  isHideClock?: boolean;
  isLoading?: boolean;
  sport?: TSport;
  teams: ITeamProps[];
  timestamp: number;
  onTeamClick?: ((teamId: number) => void) | null;
  onChange?: ((bet: IBetId[]) => void) | null;
}

export interface IMatchStatusProps {
  isBetEmpty: boolean;
  isError?: boolean;
  isMobile: boolean;
  isLoading: boolean;
  matchId: number;
}

export interface IBetId {
  id: number;
  bet: number | null;
}
