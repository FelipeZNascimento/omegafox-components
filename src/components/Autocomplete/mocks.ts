import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
