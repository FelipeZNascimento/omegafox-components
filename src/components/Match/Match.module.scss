@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  display: flex;
  flex: 1;
  flex-direction: column;
  width: 100%;
}

.matchContainer {
  cursor: pointer;
  display: flex;
  flex-direction: row;
  flex: 1;
  height: $match-container-height;
  position: relative;
  width: 100%;
  transition: filter $transition-time;

  &:hover {
    filter: brightness(90%);
    transition: filter $transition-time;
  }
}

.dateContainer {
  @include flex-aligned(center, flex-start);
  @include font-colored($color-grey5);
  @include padded(all, m);
  @include marged(right, s2);

  border-radius: $border-radius;
  border-bottom-left-radius: 0;

  flex: 0;
}

.teamContainer {
  @include font-sized(l);

  border-radius: $border-radius;
  flex: 1;
  height: $match-container-height;
  position: relative;
  overflow: hidden;
  z-index: 1;

  &Left {
    @extend .teamContainer;
    @include flex-aligned(flex-start, center);
  }
  &Right {
    @extend .teamContainer;
    @include flex-aligned(flex-end, center);
  }
}

.logoContainer {
  flex: 1;

  .logo {
    filter: drop-shadow(0px 0px 10px #222);
    position: absolute;
    top: 50%;
    z-index: $hidden-z;
    height: 90px;
    width: auto;

    &Left {
      @extend .logo;
      left: 15%;
      transform: translateX(-50%) translateY(-50%);
    }

    &Right {
      @extend .logo;
      transform: translateX(50%) translateY(-50%);
      right: 15%;
    }
  }
}

.scoreContainer {
  @include bg-rgba-colored($color-black, 0.2);
  @include flex-aligned(center, center);
  @include font-sized(l3);
  @include padded(all, m);

  flex: 0;
  height: 100%;
}

.nameContainer {
  @include font-sized(l2);
  @include padded(horizontal, m);
}

.toggle {
  @include bg-rgba-colored($color-black, 0.4);
  @include font-colored($color-white);
  @include font-sized(l2);
  @include flex-aligned(center, center);

  height: 24px;
  width: 24px;
  bottom: 0px;
  position: absolute;
  left: 0px;

  transition: filter $transition-time;
}

.expandedContainer {
  @include bg-rgba-colored($color-grey1, 0.9);
  @include flex-aligned(flex-start, center);
  @include aligned(center);
  @include shadowed(0.2);

  border-radius: $border-radius;
  border-top-left-radius: 0;
  flex-direction: column;

  &Open {
    @include padded(all, m);
    @include marged(top, s2);

    max-height: 500px;
    transition: max-height $transition-time ease-in, padding $transition-time ease-in;
  }

  &Closed {
    @include padded(all, none);
    max-height: 0;
    overflow: hidden;
    transition: max-height $transition-time ease-out, padding $transition-time ease-out;
  }
}
