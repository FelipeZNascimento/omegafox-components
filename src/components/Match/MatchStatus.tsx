import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faCircleCheck,
  faCircleDot,
  faCircleNotch,
  faCircleXmark
} from '@fortawesome/free-solid-svg-icons';

import classNames from 'classnames';
import styles from './Match.module.scss';
import { IMatchStatusProps } from '../types';
import { Tooltip } from 'components/Tooltip/Tooltip';

export const MatchStatus = ({
  isBetEmpty,
  isError = false,
  isLoading,
  isMobile
}: IMatchStatusProps) => {
  const matchStatusClassMobile = classNames(styles.matchStatusMobile, {
    [styles.matchStatusMobileLoading]: isLoading,
    [styles.matchStatusMobileCheck]: !isLoading && !isBetEmpty && !isError,
    [styles.matchStatusMobileEmpty]: !isLoading && isBetEmpty,
    [styles.matchStatusMobileError]: !isLoading && isError
  });

  const matchStatusClass = classNames(styles.matchStatus, {
    [styles.matchStatusLoading]: isLoading,
    [styles.matchStatusCheck]: !isLoading && !isBetEmpty && !isError,
    [styles.matchStatusEmpty]: !isLoading && isBetEmpty,
    [styles.matchStatusError]: !isLoading && isError
  });

  const renderTooltipText = () => {
    if (isLoading) {
      return 'Carregando...';
    } else {
      if (isError) {
        return 'Erro. Recarregue a pÃ¡gina.';
      } else if (isBetEmpty) {
        return 'Aposta incompleta.';
      } else {
        return 'Aposta completa.';
      }
    }
  };

  const renderStatusIcon = () => {
    if (isLoading) {
      return faCircleNotch;
    } else {
      if (isError) {
        return faCircleXmark;
      } else if (isBetEmpty) {
        return faCircleDot;
      } else {
        return faCircleCheck;
      }
    }
  };

  return (
    <div className={isMobile ? matchStatusClassMobile : matchStatusClass}>
      <Tooltip
        position="bottom"
        type={isError ? 'error' : 'dark'}
        text={renderTooltipText()}
      >
        <FontAwesomeIcon icon={renderStatusIcon()} />
      </Tooltip>
    </div>
  );
};
