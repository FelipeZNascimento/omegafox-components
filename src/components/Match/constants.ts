import { TBET_VALUES } from './types';
export const BET_VALUES: TBET_VALUES = {
  FULL: 5,
  HALF: 3,
  MINIMUN: 2,
  MISS: 0
};
