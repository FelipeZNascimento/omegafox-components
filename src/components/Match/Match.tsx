import { useState, useEffect } from 'react';
import { isMobile } from 'react-device-detect';

import { MatchStatus } from './MatchStatus';
import { Clock, Team } from '../index';
import { footballClock } from './mocks';
import { IMatchProps } from '../types';
import classNames from 'classnames';
import styles from './Match.module.scss';
import { IScoreId } from './types';

export const Match = ({
  betStatus = 'neutral',
  clock = footballClock,
  id,
  isBigLogo = false,
  isEditable = true,
  isError = false,
  isExpandable = false,
  isForceMobile = false,
  isLoading = false,
  timestamp,
  teams,
  expandableContent,
  onChange = null
}: IMatchProps) => {
  const [isExpanded, setIsExpanded] = useState<boolean>(false);
  const [isBetEmpty, setIsBetEmpty] = useState<boolean>(false);
  const [editedScore, setEditedScore] = useState<IScoreId[]>([]);

  useEffect(() => {
    const newScoreArray = teams.map((team) => {
      return { id: team.id, score: team.score };
    });
    setEditedScore(newScoreArray);
    setIsBetEmpty(newScoreArray.some((item) => item.score === null));
  }, [teams]);

  const matchContainerClass = classNames(styles.matchContainer, {
    [styles.matchContainerHoverable]: !isEditable,
    [styles.matchContainerExpandable]: isExpandable
  });

  const expandedContainerClass = classNames(styles.expandedContainer, {
    [styles.expandedContainerOpen]: isExpanded,
    [styles.expandedContainerClosed]: !isExpanded
  });

  const handleScoreChange = (newScore: number | null, teamId: number) => {
    const newScoreArray = editedScore.map((item) => {
      if (item.id === teamId) {
        return {
          id: item.id,
          score: newScore
        };
      }
      return item;
    });

    setEditedScore(newScoreArray);
    setIsBetEmpty(newScoreArray.some((item) => item.score === null));
    if (onChange) {
      onChange(newScoreArray);
    }
  };

  return (
    <div className={styles.container}>
      <div
        className={matchContainerClass}
        onClick={() => setIsExpanded(!isExpanded)}
      >
        {isEditable && (
          <MatchStatus
            isBetEmpty={isBetEmpty}
            isError={isError}
            isLoading={isLoading}
            isMobile={isMobile || isForceMobile}
          />
        )}

        {!isMobile && !isForceMobile && (
          <Clock
            betStatus={betStatus}
            clock={clock}
            isMatchEditable={isEditable}
            isExpandable={isExpandable}
            isExpanded={isExpanded}
            timestamp={timestamp}
          />
        )}
        {teams.map((team) => {
          const score = editedScore.find((item) => item.id === team.id)?.score;

          return (
            <Team
              key={team.id}
              align={team.align}
              colors={team.colors}
              id={team.id}
              isBigLogo={isBigLogo}
              isEditable={isEditable}
              isForceMobile={isForceMobile}
              logo={team.logo}
              matchId={id}
              name={team.name}
              nameShort={team.nameShort}
              score={score === undefined ? null : score}
              onChange={handleScoreChange}
            />
          );
        })}
      </div>
      {isExpandable && expandableContent && (
        <div className={expandedContainerClass}>{expandableContent()}</div>
      )}
    </div>
  );
};
