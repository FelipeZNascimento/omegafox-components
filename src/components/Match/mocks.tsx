import { IMatchProps, ITeamProps, TClockFootball } from '../types';
import { FOOTBALL_MATCH_STATUS } from '../Clock/constants';

const matchId = 0;
export const teamLeft: ITeamProps = {
  id: 0,
  align: 'left',
  colors: ['#FFDC02', '#19AE47'],
  isEditable: false,
  isForceMobile: false,
  logo: 'https://teamcolorcodes.com/wp-content/uploads/2021/12/Brazil-National-Football-Team-Logo-211x300.png',
  matchId: matchId,
  name: 'Brasil',
  nameShort: 'BRA',
  score: 4
};

export const teamRight: ITeamProps = {
  id: 1,
  align: 'right',
  colors: ['#FFFFFF', '#43A1D5'],
  isEditable: false,
  isForceMobile: false,
  logo: 'https://teamcolorcodes.com/wp-content/uploads/2021/12/Argentina-National-Football-Team-Logo-192x300.png',
  matchId: matchId,
  name: 'Argentina',
  nameShort: 'ARG',
  score: 0
};

export const teamLeftNoBet: ITeamProps = {
  ...teamLeft,
  score: null
};

export const teamRightNoBet: ITeamProps = {
  ...teamRight,
  score: null
};

export const footballClock: TClockFootball = {
  time: 45,
  status: FOOTBALL_MATCH_STATUS.FIRST_HALF
};

export const matchInfo: IMatchProps = {
  id: matchId,
  isExpandable: false,
  clock: footballClock,
  expandableContent: () => {
    return <p>Vila Capanema</p>;
  },
  sport: 'football',
  teams: [teamLeft, teamRight],
  timestamp: 1662682800
};

export const matchInfoNoBets: IMatchProps = {
  ...matchInfo,
  teams: [teamLeftNoBet, teamRightNoBet]
};
