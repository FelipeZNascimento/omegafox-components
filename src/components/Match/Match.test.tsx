import { render } from '@testing-library/react';

import { Match } from './Match';
import { IMatchProps } from './types';
import { footballClock, matchInfo } from './mocks';

describe('MatchComponent', () => {
  const renderComponent = ({
    clock,
    timestamp,
    teams,
    location,
    sport,
    stadium
  }: IMatchProps) =>
    render(
      <Match
        clock={clock}
        timestamp={timestamp}
        teams={teams}
        location={location}
        sport={sport}
        stadium={stadium}
      />
    );

  it('should render', () => {
    const { container } = renderComponent({
      clock: footballClock,
      location: matchInfo.location,
      sport: matchInfo.sport,
      stadium: matchInfo.stadium,
      teams: matchInfo.teams,
      timestamp: matchInfo.timestamp
    });

    expect(container).toMatchSnapshot();
  });
});
