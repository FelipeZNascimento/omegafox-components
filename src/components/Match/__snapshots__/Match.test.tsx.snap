// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchComponent should render 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="matchContainer"
    >
      <div
        class="dateContainer"
        style="background: rgb(236, 239, 241) url();"
      >
        <div
          class="toggle toggleNeutral"
        >
          +
        </div>
        <div
          class="info infoClock"
        >
          <span
            class="infoClockStatus"
          >
            1T
          </span>
          <span
            class="infoClockAnimated"
          >
            45'
          </span>
        </div>
      </div>
      <div
        class="containerLeft"
        style="color: rgb(255, 220, 2); background: rgb(25, 174, 71) url();"
      >
        <div
          class="logoContainer"
        >
          <img
            alt="logo"
            class="logoLeft"
            src="https://teamcolorcodes.com/wp-content/uploads/2021/12/Brazil-National-Football-Team-Logo-211x300.png"
          />
        </div>
        <div
          class="nameContainer"
          style="text-shadow: -1px 0 #19AE47, 0 1px #19AE47, 1px 0 #19AE47, 0 -1px #19AE47;"
        >
          Brasil
        </div>
        <div
          class="scoreContainer"
        >
          4
        </div>
      </div>
      <div
        class="containerRight"
        style="color: rgb(255, 255, 255); background: rgb(67, 161, 213) url();"
      >
        <div
          class="scoreContainer"
        >
          0
        </div>
        <div
          class="nameContainer"
          style="text-shadow: -1px 0 #43A1D5, 0 1px #43A1D5, 1px 0 #43A1D5, 0 -1px #43A1D5;"
        >
          Argentina
        </div>
        <div
          class="logoContainer"
        >
          <img
            alt="logo"
            class="logoRight"
            src="https://teamcolorcodes.com/wp-content/uploads/2021/12/Argentina-National-Football-Team-Logo-192x300.png"
          />
        </div>
      </div>
    </div>
    <div
      class="expandedContainer expandedContainerClosed"
    >
      Estádio: 
      Vila Capanema
      <br />
      Localização: 
      Curitiba, Paraná
    </div>
  </div>
</div>
`;
